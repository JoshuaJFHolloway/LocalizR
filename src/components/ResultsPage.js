import React from 'react'
import SubHeader from './SubHeader'
import SubmitButton from "./SubmitButton";
import axios from 'axios';
import Story from "./Story";


// function to drop database

const ResultsPage = props => {

  return (
    <div>
      <SubHeader
        subHeader={"Nice one, your score is " + props.score}/>
      <SubmitButton
        buttonName = {"Languages Page"}
        handleDataSubmitted = {props.handleDataSubmitted}
      />
      <SubmitButton
        buttonName = {"Retry Quiz"}
        handleDataSubmitted = {props.handleRetryQuiz}
      />
      <Story />
    </div>
  )
};

export default ResultsPage;
