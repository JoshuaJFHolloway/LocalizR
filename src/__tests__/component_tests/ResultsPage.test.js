import React from 'react';
import { shallow } from 'enzyme';
import ResultsPage from '../../components/ResultsPage';

describe('ResultsPage', () => {
  let resultspage = shallow(<ResultsPage />);

  describe('Rendering components', () => {
    it('renders correcrly', () => {
      expect(resultspage).toMatchSnapshot();
    });

    it('renders SubHeader correctly', () => {
      expect(resultspage.find('SubHeader').exists()).toBe(true);
    });
  });
});
