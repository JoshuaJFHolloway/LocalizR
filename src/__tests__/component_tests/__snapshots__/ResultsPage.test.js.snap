// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ResultsPage Rendering components renders correcrly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ResultsPage />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <SubHeader
          subHeader="Nice one, your score is undefined"
/>,
        <SubmitButton
          buttonName="Languages Page"
          handleDataSubmitted={undefined}
/>,
        <SubmitButton
          buttonName="Retry Quiz"
          handleDataSubmitted={undefined}
/>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "subHeader": "Nice one, your score is undefined",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "buttonName": "Languages Page",
          "handleDataSubmitted": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "buttonName": "Retry Quiz",
          "handleDataSubmitted": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <SubHeader
            subHeader="Nice one, your score is undefined"
/>,
          <SubmitButton
            buttonName="Languages Page"
            handleDataSubmitted={undefined}
/>,
          <SubmitButton
            buttonName="Retry Quiz"
            handleDataSubmitted={undefined}
/>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "subHeader": "Nice one, your score is undefined",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "buttonName": "Languages Page",
            "handleDataSubmitted": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "buttonName": "Retry Quiz",
            "handleDataSubmitted": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
    "disableLifecycleMethods": true,
  },
}
`;
